app/models/post.rb

```diff
@@ -30,11 +56,26 @@ def self.search(year: '', text: '', bookmarked: nil)
     posts
   end
 
-  def self.for_words(terms)
-    if terms.blank?
+  def self.filter_for_export(timeframe: 'All Time', text: '', post_type_ids: [], bookmarked: nil, with_people: '', category_ids: [])
+    posts = all
+    return [] if posts.blank?
+
+    posts = Report.timeframe_labels.include?(timeframe) ? for_timeframe(timeframe) : for_year(timeframe)
+    posts = posts.in_post_type_ids(post_type_ids) if posts.present? && post_type_ids.present?
+    posts = posts.for_words(text) if posts.present? && text.present?
+    posts = posts.for_words(with_people) if posts.present? && with_people.present?
+    posts = posts.bookmarked if posts.present? && !bookmarked.nil?
+    posts = posts.in_category_ids(category_ids) if posts.present? && category_ids.present?
+    posts = posts.by_date if posts.present?
+    posts
+  end
+
+  def self.for_words(text)
+    if text.blank?
       includes(:post_type).by_date
     else
-      includes(:post_type).where('description ILIKE ? OR given_by ILIKE ?', "%#{terms}%", "%#{terms}%").by_date
+      concat_statement = "concat_ws(' ', description, given_by)"
+      includes(:post_type).where("#{concat_statement} ILIKE ?", "%#{text}%").by_date
```

Comment:
This little refactor is satisfying to me because it scales in a nicer way than the original version does. `concat_ws` is a neat feature that postgres has, so I figured I'd use it. I chose to do it in 2 lines instead of 1 just so FutureMe won't have much to visually unpack when she comes back to this method.